src/accelerate/hooks.py

```diff
@@ -123,11 +123,21 @@ def add_hook_to_module(module: nn.Module, hook: ModelHook):
     Args:
         module (`torch.nn.Module`): The module to attach a hook to.
         hook (`ModelHook`): The hook to attach.
+        append (`bool`): Whether, if module already contains a hook, should chain the new one with SequentialHook
 
     Returns:
         `torch.nn.Module`: The same module, with the hook attached (the module is modified in place, so the result can
         be discarded).
     """
+
+    if append:
+        if hasattr(module, "_hf_hook") and (module._hf_hook is not None):
```

Comment:
Those two ifs can be grouped.
Also the second test can be simplified in `getattr(module, "_hf_hook", None) is not None`

Reply:
Done, thanks for the great idea.